mysql-test/t/mysqldump.test

```diff
@@ -2689,3 +2689,19 @@ select * from t2;
 select * from t3;
 desc t3;
 drop database d;
+
+#Bug fix for https://jira.mariadb.org/browse/MDEV-15021
```

Comment:
When adding a test case, you need to add the .result file too. Generate it with `./mtr --record mysqldump`

Also, when adding comments, it's best to structure them like so:
```
--echo #
--echo # MDEV-15021: mysqldump --tables --routines generates non importable
--echo #             dump file
--echo #
--echo # Check the order in which routines are written in the dump. Routines
--echo # need to be printed before views, in case views make use of them.
--echo #
```

Note that you should not use function names in test cases comments as function names can change and that would leave code inconsistent with the comments. Comments should explain the reasoning for the change and what the test checks.

Include the MDEV number and the title for it at the start of the comment. Usually we don't include links as links often become dead and it's extra maintenance for no reason. Just having the MDEV number will usually provide sufficient info to locate the issue in the issue tracker or its archives.